src/component/Todo.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useEffect } from "react";
+import { Form, Modal } from "react-bootstrap";
+import DateTimePicker from "react-datetime-picker";
+import { useDispatch, useSelector } from "react-redux";
+import { addTodo } from "../redux/todoSlice";
+import "bootstrap/dist/css/bootstrap.min.css";
+import "../App.css";
+import TodoData from "./TodoData";
+import moment from "moment";
+import Nav from "./Nav";
+import { TodoItem, Value } from "./Types";
+
+const Todo: React.FC = () => {
+  const [modalDisplay, setModalDisplay] = useState<boolean>(false);
+  const [inputValue, setInputValue] = useState<string>("");
+  const [timeOfTodo, onChange] = useState<Value>(moment().toDate());
+  const [error, setError] = useState<boolean>(false);
+  const [todoCompleted, setTodoCompleted] = useState<boolean>(false);
+  const dispatch = useDispatch();
+  const [items, setItems] = useState<TodoItem[]>([]);
+  const [checkInput, setCheckInput] = useState<boolean>(true);
+
+  const check = useSelector((state: any) => state.todos);
+  console.log(check);
+
+  const updateItemColors = () => {
+    const updatedItems = check.map((item: TodoItem) => ({
+      ...item,
+      color: moment(item.dateTime).isBefore(moment())
+        ? "red-dot"
+        : "purple-dot",
+    }));
+    setItems([...updatedItems]);
+  };
+
+  const handleCheck = (id: number) => {
+    const itemToUpdate = items.find((item) => item.id === id);
+    if (itemToUpdate) {
+      setTodoCompleted(!itemToUpdate.completed);
+      const updatedItems = items.map((item) =>
+        item.id === id ? { ...item, completed: !item.completed } : item
+      );
+      setItems(updatedItems);
```

Comment:
add/update todoItems all should be in redux actions and state @Rishab-innovative